front/components/providers/AI21Setup.jsx

```diff
@@ -0,0 +1,183 @@
+import { useSWRConfig } from "swr";
+import { checkProvider } from "../../lib/providers";
+
+export default function AI21Setup({ open, setOpen, config, enabled }) {
+  const { mutate } = useSWRConfig();
+
+  const [apiKey, setApiKey] = useState(config ? config.api_key : "");
+  const [testSuccessful, setTestSuccessful] = useState(false);
+  const [testRunning, setTestRunning] = useState(false);
+  const [testError, setTestError] = useState("");
+  const [enableRunning, setEnableRunning] = useState(false);
+
+  if (config && config.api_key.length > 0 && apiKey.length == 0) {
+    setApiKey(config.api_key);
+  }
+
+  const runTest = async () => {
+    setTestRunning(true);
+    setTestError("");
+    let check = await checkProvider("ai21", { api_key: apiKey });
+
+    if (!check.ok) {
+      setTestError(check.error);
+      setTestSuccessful(false);
+      setTestRunning(false);
+    } else {
+      setTestError("");
+      setTestSuccessful(true);
+      setTestRunning(false);
+    }
+  };
+
+  const handleEnable = async () => {
+    setEnableRunning(true);
+    let res = await fetch(`/api/providers/ai21`, {
+      headers: {
+        "Content-Type": "application/json",
+      },
+      method: "POST",
+      body: JSON.stringify({
+        config: JSON.stringify({
+          api_key: apiKey,
+        }),
+      }),
+    });
+    setEnableRunning(false);
+    mutate(`/api/providers`);
+    setOpen(false);
+  };
+
+  const handleDisable = async () => {
+    let res = await fetch(`/api/providers/ai21`, {
+      method: "DELETE",
+    });
+    mutate(`/api/providers`);
+    setOpen(false);
+  };
+
+  return (
+    <Transition.Root show={open} as={Fragment}>
+      <Dialog as="div" className="relative z-10" onClose={() => {}}>
+        <Transition.Child
+          as={Fragment}
+          enter="ease-out duration-300"
+          enterFrom="opacity-0"
+          enterTo="opacity-100"
+          leave="ease-in duration-200"
+          leaveFrom="opacity-100"
+          leaveTo="opacity-0"
+        >
+          <div className="fixed inset-0 bg-gray-800 bg-opacity-75 transition-opacity" />
+        </Transition.Child>
+
+        <div className="fixed inset-0 z-10 overflow-y-auto">
+          <div className="flex min-h-full items-end justify-center p-4 items-center">
+            <Transition.Child
+              as={Fragment}
+              enter="ease-out duration-300"
+              leave="ease-in duration-200"
+              leaveTo="opacity-0"
+            >
+              <Dialog.Panel className="relative transform overflow-hidden rounded-lg bg-white px-4 pt-5 pb-4 text-left shadow-xl transition-all sm:my-8 sm:w-full sm:max-w-sm lg:max-w-lg sm:p-6">
+                <div>
+                  <div className="mt-3">
+                    <Dialog.Title
+                      as="h3"
+                      className="text-lg font-medium leading-6 text-gray-900"
+                    >
+                      Setup AI21 Studio
```

Comment:
Should we remove the "Studio" and just say "AI21" here and in the provider name?

Reply:
As per IRL: let's keep it